### PR TITLE
new port: nushell

### PR DESCRIPTION
**Pull request by b4nst**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.0.1 20B50 x86_64
Xcode macOS 11.0.1 20B50 x86_64

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: shells/nushell/Portfile [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+license             MIT
+
+description         A new type of shell
+long_description    ${description}
+
+homepage            https://www.nushell.sh
+
+maintainers         {banst @b4nst} \
+                    openmaintainer
+
```
Comment (reneeotten) [inline shells/nushell/Portfile:20]: no need for extra newline

File: shells/nushell/Portfile [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+                    size    6247670
+
+destroot {
+    delete {*}[glob ${worksrcpath}/target/[cargo.rust_platform]/release/*.d]
+
+    xinstall -m 0755 ${worksrcpath}/target/[cargo.rust_platform]/release/nu ${destroot}${prefix}/bin
+    # plugins
+    xinstall -m 0755 {*}[glob ${worksrcpath}/target/[cargo.rust_platform]/release/nu_plugin_*] ${destroot}${prefix}/bin
+
+    xinstall -m 644 -W ${worksrcpath} LICENSE ${destroot}${prefix}/share/doc/${name}
```
Comment (reneeotten) [inline shells/nushell/Portfile:36]: please use 0644

**Review by reneeotten — CHANGES_REQUESTED** [review]
two minor comments; also please make sure that your commit message follows the guidelines. It should be "portname: short description", so in this case "nushell: new port, version x.y". 

Please amend or squash your commits when making these changes and rebase with current "master"; that will re-start the CI runs as well so we can see they actually pass. It failed last time with errors not related to this port, hence the rebase reauesr.

File: shells/nushell/Portfile [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cargo 1.0
+
+github.setup        nushell nushell 0.32.0
+revision            0
+categories          shells
+platforms           darwin
+license             MIT
+maintainers         {banst @b4nst} openmaintainer
```
Comment (reneeotten) [inline shells/nushell/Portfile:12]: sorry, one more - this line is actually not correct; your e-mail address should be listed as ```example.org:user```

Comment (reneeotten) [reply shells/nushell/Portfile:12]: also, please change your commit message as I mentioned above

Comment (b4nst) [reply shells/nushell/Portfile:12]: Oops sorry I didn't catch the message about the commit. Will do

**Comment by reneeotten** [comment]
thank you for the new port @b4nst !

**Review by reneeotten — APPROVED** [review]

**Pull request merged**